buildSrc/src/test/java/org/elasticsearch/gradle/DistributionDownloadPluginIT.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import org.elasticsearch.gradle.test.GradleIntegrationTestCase;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.head;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+public class DistributionDownloadPluginIT extends GradleIntegrationTestCase {
+    // TODO: check reuse of root task across projects MOVE TO UNIT TEST
+    // TODO: make extracted assert also do file assert, to reduce separate test builds
+    // TODO: future: check integ-test-zip to maven, snapshots to snapshot service for external project
+
+    private static final Pattern DISTRO_FILE_LOGLINE = Pattern.compile("DISTRO FILE: (.*)");
+    private static final Pattern DISTRO_EXTRACTED_LOGLINE = Pattern.compile("DISTRO EXTRACTED: (.*)");
+
+    public void testCurrent() throws Exception {
```

Comment:
I think we should split these tests up and have a separate one for the file distro and extracted one. These tests currently run two separate builds, either of which could fail for a different reason. I think it makes more sense to have more discrete tests.

Reply:
These tests are quite slow, relatively, so i want to reduce the number of testkit builds we run. Currently the way this is structured, we run one within each assert method. I have a TODO at the top of this file to change that so the assertion for expanded checks both, because if we can use expanded, the file version should also exist.

Reply:
Right, this test contains two assertions, so two testkit builds. Splitting this into two tests won't incur any additional build executions so I think it makes sense to split it up, unless we combine both assertions into a single build execution.